packages/components/resource-pages/list-page.tsx

```diff
@@ -0,0 +1,246 @@
+        id: tableColumnInfo[3].id
+      },
+      {
+        title: "",
+        props: {
+          className: tableColumnInfo[4].className
+        },
+        id: tableColumnInfo[4].id
+      }
+    ],
+    [t]
+  );
+
+  const [columns] = useActiveColumns({
+    columns: tableColumns,
+    showNamespaceOverride: false,
+    columnManagementID: null
+  });
+  return (
+    <VirtualizedTable
+      {...props}
+      aria-label={t("Resource Page")}
+      columns={columns}
+      Row={RowRenderer}
+    />
+  );
+};
+
+type CustomData = {
+  launchModal: LaunchModal;
+  resourceModel: K8sModel;
+  kebabActions?: (t: TFunction) => { [key: string]: { value: string } };
+};
+
+const RowRenderer: React.FC<RowProps<K8sResourceCommon, CustomData>> = ({
+  obj,
+  activeColumnIDs,
+  rowData
+}) => {
+  const { resourceModel, launchModal, kebabActions } = rowData;
+  const name = getName(obj);
+  const provider = getProvider(obj);
+  return (
+    <>
+      <TableData {...tableColumnInfo[0]} activeColumnIDs={activeColumnIDs}>
+        {name}
+      </TableData>
+      <TableData {...tableColumnInfo[1]} activeColumnIDs={activeColumnIDs}>
+        <OperandStatus operand={obj} />
+      </TableData>
+      <TableData {...tableColumnInfo[2]} activeColumnIDs={activeColumnIDs}>
+        {provider}
+      </TableData>
+      <TableData {...tableColumnInfo[3]} activeColumnIDs={activeColumnIDs}>
+        {2 + " Buckets"}
+      </TableData>
+      <TableData {...tableColumnInfo[4]} activeColumnIDs={activeColumnIDs}>
+        <Kebab
+          launchModal={launchModal}
+          extraProps={{ resource: obj, resourceModel: resourceModel }}
+          customKebabItems={kebabActions}
+        />
+      </TableData>
+    </>
+  );
+};
+
+type GenericListPageProps = {
+  actions?: ModalMap;
+  resourceModel: K8sModel;
+  kebabActions?: CustomData["kebabActions"];
+};
+
+const GenericListPage: React.FC<GenericListPageProps> = ({
+  actions,
+  resourceModel,
+  kebabActions
+}) => {
+  const { t } = useTranslation();
+  const [Modal, modalProps, launchModal] = useModalLauncher(actions);
+
+  const resource = React.useMemo(
+    () => ({
+      kind: referenceForModel(resourceModel),
+      ns: DATA_FEDERATION_NAMESPACE,
+      isList: true
+    }),
+    [resourceModel]
+  );
+  const [resources, loaded, loadError] =
+    useK8sWatchResource<K8sResourceCommon[]>(resource);
+  const [data, filteredData, onFilterChange] = useListPageFilter(resources);
+  const new_filtered_array = (filteredData as NamespaceStoreKind[])
+    .filter((a) => {
+      return a.metadata.namespace === DATA_FEDERATION_NAMESPACE;
+    })
+    .map((a) => {
+      return {
+        ...a,
+        provider: Object.keys(NOOBAA_TYPE_MAP).find(
+          (key) => NOOBAA_TYPE_MAP[key] === a.spec.type
+        )
+      };
+    });
+  const createLink = `/dfr/resource/${referenceForModel(
+    resourceModel
+  )}/create/~new`;
+
+  return (
+    <>
+      <Modal {...modalProps} />
+      <div>
```

Comment:
instead of this `div` can u check once how it will show if we use `ListPageHeader` here...
maybe that way we don't need to add `margin-top: 1.5rem` to the CSS.

Reply:
tried it is coming with some additional padding and inside that I am not able to move the 2 object to extreme left as float is not woking